actix-http/src/body/body.rs

```diff
@@ -33,29 +34,60 @@ impl AnyBody {
         Self::Bytes(Bytes::new())
     }
 
-    /// Create body from slice (copy)
+    /// Create boxed body from generic message body.
+    pub fn new_boxed<B>(body: B) -> Self
+    where
+        B: MessageBody + 'static,
+        B::Error: Into<Box<dyn StdError + 'static>>,
+    {
+        Self::Body(BoxBody::from_body(body))
+    }
+
+    /// Constructs new `AnyBody` instance from a slice of bytes by copying it.
+    ///
+    /// If your bytes container is owned, it may be cheaper to use a `From` impl.
+    pub fn copy_from_slice(s: &[u8]) -> Self {
+        Self::Bytes(Bytes::copy_from_slice(s))
+    }
+
+    #[doc(hidden)]
+    #[deprecated(since = "4.0.0", note = "Renamed to `copy_from_slice`.")]
     pub fn from_slice(s: &[u8]) -> Self {
         Self::Bytes(Bytes::copy_from_slice(s))
     }
+}
 
+impl<B> AnyBody<B>
+where
+    B: MessageBody + 'static,
+    B::Error: Into<Box<dyn StdError + 'static>>,
+{
     /// Create body from generic message body.
-    pub fn from_message<B>(body: B) -> Self
-    where
-        B: MessageBody + 'static,
-        B::Error: Into<Box<dyn StdError + 'static>>,
-    {
-        Self::Stream(BoxAnyBody::from_body(body))
+    pub fn new(body: B) -> Self {
+        Self::Body(body)
+    }
+
+    pub fn into_boxed(self) -> AnyBody {
+        match self {
+            AnyBody::None => AnyBody::new_boxed(()),
```

Comment:
```rust
match self {
    Self::None => AnyBody::None,
    Self::Bytes(bytes) => AnyBody::Bytes(bytes)
}
```
This would keep all response with fixed in memory body unboxed.